http/http-api/src/main/java/com/okta/commons/http/config/HttpClientConfiguration.java

```diff
@@ -36,6 +43,17 @@
     private Proxy proxy;
     private int retryMaxElapsed = 0;
     private int retryMaxAttempts = 0;
+    private final Map<String, String> requestExecutorParams = new HashMap<>();
+
+    public static final int DEFAULT_MAX_CONNECTIONS_PER_ROUTE = Integer.MAX_VALUE/2;
+    public static final int DEFAULT_MAX_CONNECTIONS_TOTAL = Integer.MAX_VALUE;
+    public static final int DEFAULT_CONNECTION_VALIDATION_INACTIVITY = 2000; // 2sec
+    public static final int DEFAULT_CONNECTION_TIME_TO_LIVE = 5 * 1000 * 60; // 5 minutes
+
+    public static final String MAX_CONNECTIONS_PER_ROUTE_PROPERTY_KEY = "com.okta.sdk.impl.http.httpclient.HttpClientRequestExecutor.connPoolControl.maxPerRoute";
+    public static final String MAX_CONNECTIONS_TOTAL_PROPERTY_KEY = "com.okta.sdk.impl.http.httpclient.HttpClientRequestExecutor.connPoolControl.maxTotal";
+    public static final String CONNECTION_VALIDATION_PROPERTY_KEY = "com.okta.sdk.impl.http.httpclient.HttpClientRequestExecutor.connPoolControl.validateAfterInactivity";
+    public static final String CONNECTION_TIME_TO_LIVE_PROPERTY_KEY = "com.okta.sdk.impl.http.httpclient.HttpClientRequestExecutor.connPoolControl.timeToLive";
```

Comment:
I think there was some miss communication here.

The goal is to pass all values from `okta.client.requestExecutor` as a Map to the underlying implementation.
It looks like requestExecutorParams would do this.  These typed methods are going to be confusing to anyone using the OkHttp implementation which does NOT support these values.

IMHO, we should simplify this and the corresponding change in the SDK, the parsing of the properties can remain in the HttpClientRequestExcutor (as these properties are specific to that implementation)

I think part of the misunderstanding is that this class is named `HttpClientConfiguration` this is NOT specific to the `HttpClient` implementation, its generic to any "http client"

Reply:
updated